### PR TITLE
add integration test capability

### PR DESCRIPTION
**Pull request by jsotobroad**

### CODE REVIEW

File: src/bin/travis/test.sh [hunk 1]
```diff
@@ -3,19 +3,27 @@
...
-elif [ "$BUILD_TYPE" = "centaurTes" ]; then
-    "${SCRIPT_DIR}"/testCentaurTes.sh
-elif [ "$BUILD_TYPE" = "centaurLocal" ]; then
-    "${SCRIPT_DIR}"/testCentaurLocal.sh
-elif [ "$BUILD_TYPE" = "sbt" ]; then
-    "${SCRIPT_DIR}"/testSbt.sh
-elif [ "$BUILD_TYPE" = "checkPublish" ]; then
-    "${SCRIPT_DIR}"/testCheckPublish.sh
+# $TRAVIS_EVENT_TYPE will be cron is this build was initiated by a cron job
+if [ "$TRAVIS_EVENT_TYPE" != "cron" ]; then
```
Comment (mcovarr) [inline src/bin/travis/test.sh]: is there something that can positively be tested for here rather than `!= "cron"`?

File: src/bin/travis/test.sh [hunk 2]
```diff
@@ -3,19 +3,27 @@
...
+        echo "Unknown BUILD_TYPE: '$BUILD_TYPE'"
+        exit 1
+    fi
 else
-    echo "Unknown BUILD_TYPE: '$BUILD_TYPE'"
-    exit 1
+    if [ "$BUILD_TYPE" = "centaurJes" ]; then
+        "${SCRIPT_DIR}"/testCentaurJes.sh -i
+    else
+        exit 0
```
Comment (mcovarr) [inline src/bin/travis/test.sh]: should this be exiting happily in this case?

**Comment by mcovarr** [comment]
:+1:

File: src/bin/travis/test.sh (hunk 1 above)
Comment (jsotobroad) [inline src/bin/travis/test.sh]: The possible values are `push, pull_request, api, cron`.  Should i put push/pull/api in one block and cron in another?

File: src/bin/travis/test.sh [hunk 3]
```diff
@@ -4,18 +4,32 @@ set -e
...
-    "${SCRIPT_DIR}"/testCentaurLocal.sh
-elif [ "$BUILD_TYPE" = "sbt" ]; then
-    "${SCRIPT_DIR}"/testSbt.sh
-elif [ "$BUILD_TYPE" = "checkPublish" ]; then
-    "${SCRIPT_DIR}"/testCheckPublish.sh
-else
-    echo "Unknown BUILD_TYPE: '$BUILD_TYPE'"
-    exit 1
+if [ "$TRAVIS_EVENT_TYPE" = "push" ] || [ "$TRAVIS_EVENT_TYPE" = "pull_request" ] || [ "$TRAVIS_EVENT_TYPE" = "api" ]; then
+    # BUILD_TYPE is coming in from the Travis build matrix
```
Comment (mcovarr) [inline src/bin/travis/test.sh]: these bash wizards have at least a couple suggestions for ways to DRY this with `case`s or pattern matches:

https://stackoverflow.com/questions/21157435/bash-string-compare-to-multiple-correct-values

File: src/bin/travis/test.sh [hunk 4]
```diff
@@ -4,18 +4,48 @@ set -e
...
-elif [ "$BUILD_TYPE" = "sbt" ]; then
-    "${SCRIPT_DIR}"/testSbt.sh
-elif [ "$BUILD_TYPE" = "checkPublish" ]; then
-    "${SCRIPT_DIR}"/testCheckPublish.sh
-else
-    echo "Unknown BUILD_TYPE: '$BUILD_TYPE'"
-    exit 1
-fi
+# $TRAVIS_EVENT_TYPE will be cron if this build was initiated by a cron job
+case "$TRAVIS_EVENT_TYPE" in
```
Comment (jsotobroad) [inline src/bin/travis/test.sh]: is this the right amount of 😡  (how is there no :magic: or :wizard:)

Comment (mcovarr) [inline src/bin/travis/test.sh]: beauteous!  :+1: :+1: :+1:

**Comment by geoffjentry** [comment]
Make sure to run all your bash through `shellcheck.net`, otherwise 👍

File: .travis.yml [hunk 5]
```diff
@@ -37,3 +38,12 @@ deploy:
   script: src/bin/travis/publishRelease.sh
   on:
     tags: true
+notifications:
+  slack:
+    rooms:
+      - secure: B5KYcnhk/ujAUWlHsjzP7ROLm6MtYhaGikdYf6JYINovhMbVKnZCTlZEy7rqT3L2T5uJ25iefD500VQGk1Gn7puQ1sNq50wqjzQaj20PWEiBwoWalcV/nKBcQx1TyFT13LJv8fbFnVPxFCkC3YXoHedx8qAhDs8GH/tT5J8XOC8=
```
Comment (Horneth) [inline .travis.yml]: Is it ok to have this stuff here ?

Comment (jsotobroad) [inline .travis.yml]: Yes this is the encrypted version as recommended in https://docs.travis-ci.com/user/notifications/#Configuring-slack-notifications

**Pull request merged**